sources/packages/api/src/database/entities/user.model.ts

```diff
@@ -1,35 +1,91 @@
-import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from "typeorm";
+import { ColumnNames, TableNames } from "../constant";
 import { BaseModel } from "./base.model";
+import { IdentityProviders } from "./identity-providers.type";
 
-@Entity({ name: "users" })
+@Entity({ name: TableNames.User })
 export class User extends BaseModel {
   @PrimaryGeneratedColumn()
   id: number;
-
+  /**
+   * Unique user name generated by Keyclock. Usually in
+   * a form of a guid and a suffix like @bceid, @idir, @bcsc.
+   */
   @Column({
     name: "user_name",
     nullable: false,
   })
   userName: string;
-
+  /**
+   * User email.
+   */
   @Column({
     name: "email",
     nullable: false,
   })
   email: string;
-
+  /**
+   * First name or given names of the user.
+   */
   @Column({
     name: "first_name",
   })
   firstName: string;
-
+  /**
+   * Last name or the family name of the user.
+   */
   @Column({
     name: "last_name",
   })
   lastName: string;
-
+  /**
+   * Indicates if the user is active in the system or should
+   * be blocked from executing the login.
+   */
   @Column({
     name: "is_active",
   })
   isActive: boolean;
+  /**
+   * Identity Provider (IDP) used by the user to authenticate
+   * to the system (e.g. BCeID, BCSC, IDIR).
+   */
+  @Column({
+    name: "identity_provider_type",
+    nullable: false,
+  })
+  identityProviderType: IdentityProviders;
+  // !creator and modifier properties are directly added here(instead of inheriting
```

Comment:
inheriting is causing cyclic dependencies?

Reply:
Yes, the `RecordDataModel` depends on the `User` and vice-versa.